In [1]:
import pandas as pd
import numpy as np

In [2]:
df=pd.read_csv('Feature Eng processed Files/Linkedin_Job_Details_and_Descriptions_26-Sep-2023-to-26-Nov-202311737-jobs_20-features_685-pages_Feature-Eng-processed_v5.csv')
df

Job Title  \
0                              LinkedIn Data Extractor   
1                                 It Management Intern   
2                               SQL Database Developer   
3                      Business Intelligence Developer   
4                           Web Developer - Internship   
...                                                ...   
11732                                  Unity Developer   
11733  Manager – Growth Chat & SMS (Digital Marketing)   
11734                  Information Security Specialist   
11735        FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
11736                              Community Associate   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
11732  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
11733  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   
11734  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
11735  Level 3 BOS · Karachi Division, Sindh, Pakista...   
11736  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
11732                         On-site\n\n \nFull-time   
11733  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11734       On-site\n\n \nFull-time\n\n \nEntry level   
11735  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11736                         On-site\n\n \nFull-time   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11732  https://www.linkedin.com/jobs/view/3767951886   
11733  https://www.linkedin.com/jobs/view/3767355025   
11734  https://www.linkedin.com/jobs/view/3772530977   
11735  https://www.linkedin.com/jobs/view/3771943212   
11736  https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
0                                        11-50 employees  3727054368   
1                                        11-50 employees  3723395858   
2                                       51-200 employees  3727034665   
3                                        11-50 employees  3727029273   
4                    Mobeen Naqvi is hiring for this job  3727034696   
...                                                  ...         ...   
11732                                     1-10 employees  3767951886   
11733                   5,001-10,000 employees · Banking  3767355025   
11734   51-200 employees · IT Services and IT Consulting  3772530977   
11735  201-500 employees · Outsourcing and Offshoring...  3771943212   
11736                                    11-50 employees  3767944844   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11732  https

In [3]:
df_3=df.copy()

In [4]:
df_4=df_4=df_3.dropna(subset=['Job Desception'])
df_4

Job Title  \
0                              LinkedIn Data Extractor   
1                                 It Management Intern   
2                               SQL Database Developer   
3                      Business Intelligence Developer   
4                           Web Developer - Internship   
...                                                ...   
11732                                  Unity Developer   
11733  Manager – Growth Chat & SMS (Digital Marketing)   
11734                  Information Security Specialist   
11735        FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
11736                              Community Associate   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
11732  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
11733  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   
11734  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
11735  Level 3 BOS · Karachi Division, Sindh, Pakista...   
11736  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
11732                         On-site\n\n \nFull-time   
11733  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11734       On-site\n\n \nFull-time\n\n \nEntry level   
11735  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11736                         On-site\n\n \nFull-time   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11732  https://www.linkedin.com/jobs/view/3767951886   
11733  https://www.linkedin.com/jobs/view/3767355025   
11734  https://www.linkedin.com/jobs/view/3772530977   
11735  https://www.linkedin.com/jobs/view/3771943212   
11736  https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
0                                        11-50 employees  3727054368   
1                                        11-50 employees  3723395858   
2                                       51-200 employees  3727034665   
3                                        11-50 employees  3727029273   
4                    Mobeen Naqvi is hiring for this job  3727034696   
...                                                  ...         ...   
11732                                     1-10 employees  3767951886   
11733                   5,001-10,000 employees · Banking  3767355025   
11734   51-200 employees · IT Services and IT Consulting  3772530977   
11735  201-500 employees · Outsourcing and Offshoring...  3771943212   
11736                                    11-50 employees  3767944844   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11732  https

In [5]:
df_4.shape

(11706, 20)

In [6]:
df_4_uncomplete_desc_IDs=[]
for i in range(0,len(df_4.iloc[:])):
    if len(df_4.iloc[i]['Job Desception'])<20:
        df_4_uncomplete_desc_IDs.append(df_4.iloc[i]['Job ID'])    

In [7]:
df_4_uncomplete_desc_IDs
len(df_4_uncomplete_desc_IDs)

1125

In [8]:
df_4_uncomp_desc=df_4[df_4['Job ID'].isin(df_4_uncomplete_desc_IDs)]
df_4_uncomp_desc

Job Title  \
23              Digital Marketing Strategist and Planner   
136                        Software Engineer-Integration   
400                                      DevOps Engineer   
433                              Real Estate Sales Agent   
440                                Google Ads Specialist   
...                                                  ...   
11731  Senior Consultant- Business Consulting- Intern...   
11732                                    Unity Developer   
11734                    Information Security Specialist   
11735          FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
11736                                Community Associate   

                                                Job Info  \
23     Prime Human Resource Services (Pvt) Ltd · Karā...   
136    People (Professional Employers Pvt Ltd) · Isla...   
400    TecBrix · Lahore, Punjab, Pakistan (Remote)  \...   
433    Fajar Real Estate · Lahore, Punjab, Pakistan (...   
440    RT WEB Solutions · Lahore District, Punjab, Pa...   
...                                                  ...   
11731  EY in Pakistan  · Karachi Division, Sindh, Pak...   
11732  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
11734  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
11735  Level 3 BOS · Karachi Division, Sindh, Pakista...   
11736  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
23                                          Full-time   
136                                         Full-time   
400                                         Full-time   
433                                         Full-time   
440                                         Full-time   
...                                               ...   
11731  Hybrid\n\n \nInternship\n\n \nMid-Senior level   
11732                         On-site\n\n \nFull-time   
11734       On-site\n\n \nFull-time\n\n \nEntry level   
11735  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11736                         On-site\n\n \nFull-time   

                                              Job Link_x  \
23         https://www.linkedin.com/jobs/view/3727011545   
136        https://www.linkedin.com/jobs/view/3727002795   
400    https://www.linkedin.com/jobs/view/3727012776/...   
433    https://www.linkedin.com/jobs/view/3727005694/...   
440    https://www.linkedin.com/jobs/view/3727058923/...   
...                                                  ...   
11731      https://www.linkedin.com/jobs/view/3767944882   
11732      https://www.linkedin.com/jobs/view/3767951886   
11734      https://www.linkedin.com/jobs/view/3772530977   
11735      https://www.linkedin.com/jobs/view/3771943212   
11736      https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
23                                     10,001+ employees  3727011545   
136                               5,001-10,000 employees  3727002795   
400                                      11-50 employees  3727012776   
433                                      11-50 employees  3727005694   
440                                      11-50 employees  3727058923   
...                                                  ...         ...   
11731                 1,001-5,000 employees · Accounting  3767944882   
11732                                     1-10 employees  3767951886   
11734   51-200 employees · IT Services and IT Consulting  3772530977   
11735  201-500 employees · Outsourcing and Offshoring...  3771943212   
11736                                    11-50 employees  3767944844   

                                          Job Link_y Job Desception  \
23     https://www.linkedin.com/jobs/view/3727011545  About the job   
136    https://www.linkedin.com/jobs/view/3727002795  About the job   
400    https://www.linkedin.com/jobs/view/3727012776  About the job   
433    https://www.linkedin.com/jobs/view/37270056

In [9]:
df_5=df_4.drop(df_4_uncomp_desc.index)
df_5

Job Title  \
0                                LinkedIn Data Extractor   
1                                   It Management Intern   
2                                 SQL Database Developer   
3                        Business Intelligence Developer   
4                             Web Developer - Internship   
...                                                  ...   
11727                 Technical Sales Support Specialist   
11728                           YouTube Channel Producer   
11729  Social Media Content Creator - Night Shift wit...   
11730                                Design Sales agents   
11733    Manager – Growth Chat & SMS (Digital Marketing)   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
11727  BUZZMAX.CA · Karāchi, Sindh, Pakistan  \n1 wee...   
11728  Senew Tech · Lahore District, Punjab, Pakistan...   
11729  Dymentions Pakistan · Karāchi, Sindh, Pakistan...   
11730  Self-employed · Pakistan  \n1 week ago\n\n · 1...   
11733  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
11727                         On-site\n\n \nFull-time   
11728                         On-site\n\n \nFull-time   
11729       On-site\n\n \nFull-time\n\n \nEntry level   
11730                          Remote\n\n \nPart-time   
11733  On-site\n\n \nFull-time\n\n \nMid-Senior level   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11727  https://www.linkedin.com/jobs/view/3765162094   
11728  https://www.linkedin.com/jobs/view/3767643794   
11729  https://www.linkedin.com/jobs/view/3769970428   
11730  https://www.linkedin.com/jobs/view/3766852400   
11733  https://www.linkedin.com/jobs/view/3767355025   

                                   employees      Job ID  \
0                            11-50 employees  3727054368   
1                            11-50 employees  3723395858   
2                           51-200 employees  3727034665   
3                            11-50 employees  3727029273   
4        Mobeen Naqvi is hiring for this job  3727034696   
...                                      ...         ...   
11727                        11-50 employees  3765162094   
11728                    501-1,000 employees  3767643794   
11729   11-50 employees · Marketing Services  3769970428   
11730  Shayan Tauqeer is hiring for this job  3766852400   
11733       5,001-10,000 employees · Banking  3767355025   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11727  https://www.linkedin.com/jobs/view/3765162094   
11728  https://www.linkedin.com/jobs/view/3767643794   
11729  https://www.lin

#### using commen keywords used in IT and Tech

In [10]:
it_tech_keywords= ['Software', 'Computer', 'Digital', 'Programming', 
                   'Data', 'Freelance', 'Web', 'Cloud', 'Graphic', 
                   'Game', 'Information technology', 'Mobile', 'Email', 
                   'Virtual', 'video', 'technology', 'ecommerce', 'network']
it_tech_keywords
len(it_tech_keywords)

filtered_df_5_desc = df_5[df_5['Job Desception'].apply(lambda desc: any(keyword.lower() in desc.lower() for keyword in it_tech_keywords))]
filtered_df_5_desc

Job Title  \
0                                LinkedIn Data Extractor   
1                                   It Management Intern   
2                                 SQL Database Developer   
3                        Business Intelligence Developer   
4                             Web Developer - Internship   
...                                                  ...   
11726                 Sales Business Development Manager   
11727                 Technical Sales Support Specialist   
11728                           YouTube Channel Producer   
11729  Social Media Content Creator - Night Shift wit...   
11733    Manager – Growth Chat & SMS (Digital Marketing)   

                                                Job Info  \
0            EcomFleet · Pakistan (Remote)  \n1 week ago   
1      Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2      Murkez Technologies · Lahore, Punjab, Pakistan...   
3      DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4      ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                  ...   
11726  EComia Global · Rawalpindi, Punjab, Pakistan  ...   
11727  BUZZMAX.CA · Karāchi, Sindh, Pakistan  \n1 wee...   
11728  Senew Tech · Lahore District, Punjab, Pakistan...   
11729  Dymentions Pakistan · Karāchi, Sindh, Pakistan...   
11733  Bank Alfalah Limited · Karāchi, Sindh, Pakista...   

                                             Job Type  \
0                                           Full-time   
1                                           Full-time   
2                                           Full-time   
3                                           Full-time   
4                                           Full-time   
...                                               ...   
11726                         On-site\n\n \nFull-time   
11727                         On-site\n\n \nFull-time   
11728                         On-site\n\n \nFull-time   
11729       On-site\n\n \nFull-time\n\n \nEntry level   
11733  On-site\n\n \nFull-time\n\n \nMid-Senior level   

                                          Job Link_x  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11726  https://www.linkedin.com/jobs/view/3767634192   
11727  https://www.linkedin.com/jobs/view/3765162094   
11728  https://www.linkedin.com/jobs/view/3767643794   
11729  https://www.linkedin.com/jobs/view/3769970428   
11733  https://www.linkedin.com/jobs/view/3767355025   

                                  employees      Job ID  \
0                           11-50 employees  3727054368   
1                           11-50 employees  3723395858   
2                          51-200 employees  3727034665   
3                           11-50 employees  3727029273   
4       Mobeen Naqvi is hiring for this job  3727034696   
...                                     ...         ...   
11726                       11-50 employees  3767634192   
11727                       11-50 employees  3765162094   
11728                   501-1,000 employees  3767643794   
11729  11-50 employees · Marketing Services  3769970428   
11733      5,001-10,000 employees · Banking  3767355025   

                                          Job Link_y  \
0      https://www.linkedin.com/jobs/view/3727054368   
1      https://www.linkedin.com/jobs/view/3723395858   
2      https://www.linkedin.com/jobs/view/3727034665   
3      https://www.linkedin.com/jobs/view/3727029273   
4      https://www.linkedin.com/jobs/view/3727034696   
...                                              ...   
11726  https://www.linkedin.com/jobs/view/3767634192   
11727  https://www.linkedin.com/jobs/view/3765162094   
11728  https://www.linkedin.com/jo

In [11]:
filtered_df_5_desc['Job Title'].unique()

array(['LinkedIn Data Extractor', 'It Management Intern',
       'SQL Database Developer', ...,
       'Technical Sales Support Specialist',
       'Social Media Content Creator - Night Shift with Flexible Hours',
       'Manager – Growth Chat & SMS (Digital Marketing)'], dtype=object)

In [12]:
len(filtered_df_5_desc['Job Title'].unique()),len(df_3['Job Title'].unique())

(5025, 6442)

### dealing Job titles where jobs desc is abnormal

In [13]:
len(df_3)-len(df_5) # there are 97 rows where Jobs desc= abnormal

1156

##### approach:1

In [14]:
df_3.drop(df_5.index)

Job Title  \
23              Digital Marketing Strategist and Planner   
48                                      Research Analyst   
136                        Software Engineer-Integration   
400                                      DevOps Engineer   
433                              Real Estate Sales Agent   
...                                                  ...   
11731  Senior Consultant- Business Consulting- Intern...   
11732                                    Unity Developer   
11734                    Information Security Specialist   
11735          FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
11736                                Community Associate   

                                                Job Info  \
23     Prime Human Resource Services (Pvt) Ltd · Karā...   
48     Indian School of Business · Hyderabad, Sindh, ...   
136    People (Professional Employers Pvt Ltd) · Isla...   
400    TecBrix · Lahore, Punjab, Pakistan (Remote)  \...   
433    Fajar Real Estate · Lahore, Punjab, Pakistan (...   
...                                                  ...   
11731  EY in Pakistan  · Karachi Division, Sindh, Pak...   
11732  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
11734  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
11735  Level 3 BOS · Karachi Division, Sindh, Pakista...   
11736  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
23                                          Full-time   
48                       Full-time · Mid-Senior level   
136                                         Full-time   
400                                         Full-time   
433                                         Full-time   
...                                               ...   
11731  Hybrid\n\n \nInternship\n\n \nMid-Senior level   
11732                         On-site\n\n \nFull-time   
11734       On-site\n\n \nFull-time\n\n \nEntry level   
11735  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11736                         On-site\n\n \nFull-time   

                                              Job Link_x  \
23         https://www.linkedin.com/jobs/view/3727011545   
48         https://www.linkedin.com/jobs/view/3723336007   
136        https://www.linkedin.com/jobs/view/3727002795   
400    https://www.linkedin.com/jobs/view/3727012776/...   
433    https://www.linkedin.com/jobs/view/3727005694/...   
...                                                  ...   
11731      https://www.linkedin.com/jobs/view/3767944882   
11732      https://www.linkedin.com/jobs/view/3767951886   
11734      https://www.linkedin.com/jobs/view/3772530977   
11735      https://www.linkedin.com/jobs/view/3771943212   
11736      https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
23                                     10,001+ employees  3727011545   
48                501-1,000 employees · Higher Education  3723336007   
136                               5,001-10,000 employees  3727002795   
400                                      11-50 employees  3727012776   
433                                      11-50 employees  3727005694   
...                                                  ...         ...   
11731                 1,001-5,000 employees · Accounting  3767944882   
11732                                     1-10 employees  3767951886   
11734   51-200 employees · IT Services and IT Consulting  3772530977   
11735  201-500 employees · Outsourcing and Offshoring...  3771943212   
11736                                    11-50 employees  3767944844   

                                          Job Link_y Job Desception  \
23     https://www.linkedin.com/jobs/view/3727011545  About the job   
48     https://www.linkedin.com/jobs/view/3723336007            NaN   
136    https://www.linkedin.com/jobs/view/3727002795  About the job   
400    https://www.linkedin.com/jobs/view/37270127

##### Approach:2

In [34]:
# FUTURE TASK
# df_3[df_3['Job ID'].isin(df_4_uncomplete_desc_IDs+null_desc_iDS)]

#### getting df_3_abnormal

In [15]:
df_3_abnormal=df_3.drop(df_5.index)
df_3_abnormal

Job Title  \
23              Digital Marketing Strategist and Planner   
48                                      Research Analyst   
136                        Software Engineer-Integration   
400                                      DevOps Engineer   
433                              Real Estate Sales Agent   
...                                                  ...   
11731  Senior Consultant- Business Consulting- Intern...   
11732                                    Unity Developer   
11734                    Information Security Specialist   
11735          FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
11736                                Community Associate   

                                                Job Info  \
23     Prime Human Resource Services (Pvt) Ltd · Karā...   
48     Indian School of Business · Hyderabad, Sindh, ...   
136    People (Professional Employers Pvt Ltd) · Isla...   
400    TecBrix · Lahore, Punjab, Pakistan (Remote)  \...   
433    Fajar Real Estate · Lahore, Punjab, Pakistan (...   
...                                                  ...   
11731  EY in Pakistan  · Karachi Division, Sindh, Pak...   
11732  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   
11734  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
11735  Level 3 BOS · Karachi Division, Sindh, Pakista...   
11736  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
23                                          Full-time   
48                       Full-time · Mid-Senior level   
136                                         Full-time   
400                                         Full-time   
433                                         Full-time   
...                                               ...   
11731  Hybrid\n\n \nInternship\n\n \nMid-Senior level   
11732                         On-site\n\n \nFull-time   
11734       On-site\n\n \nFull-time\n\n \nEntry level   
11735  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11736                         On-site\n\n \nFull-time   

                                              Job Link_x  \
23         https://www.linkedin.com/jobs/view/3727011545   
48         https://www.linkedin.com/jobs/view/3723336007   
136        https://www.linkedin.com/jobs/view/3727002795   
400    https://www.linkedin.com/jobs/view/3727012776/...   
433    https://www.linkedin.com/jobs/view/3727005694/...   
...                                                  ...   
11731      https://www.linkedin.com/jobs/view/3767944882   
11732      https://www.linkedin.com/jobs/view/3767951886   
11734      https://www.linkedin.com/jobs/view/3772530977   
11735      https://www.linkedin.com/jobs/view/3771943212   
11736      https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
23                                     10,001+ employees  3727011545   
48                501-1,000 employees · Higher Education  3723336007   
136                               5,001-10,000 employees  3727002795   
400                                      11-50 employees  3727012776   
433                                      11-50 employees  3727005694   
...                                                  ...         ...   
11731                 1,001-5,000 employees · Accounting  3767944882   
11732                                     1-10 employees  3767951886   
11734   51-200 employees · IT Services and IT Consulting  3772530977   
11735  201-500 employees · Outsourcing and Offshoring...  3771943212   
11736                                    11-50 employees  3767944844   

                                          Job Link_y Job Desception  \
23     https://www.linkedin.com/jobs/view/3727011545  About the job   
48     https://www.linkedin.com/jobs/view/3723336007            NaN   
136    https://www.linkedin.com/jobs/view/3727002795  About the job   
400    https://www.linkedin.com/jobs/view/37270127

In [16]:
df_3_abnormal['Job Title']

23                Digital Marketing Strategist and Planner
48                                        Research Analyst
136                          Software Engineer-Integration
400                                        DevOps Engineer
433                                Real Estate Sales Agent
                               ...                        
11731    Senior Consultant- Business Consulting- Intern...
11732                                      Unity Developer
11734                      Information Security Specialist
11735            FUNCTIONAL CONSULTANT ORACLE FUSION - HCM
11736                                  Community Associate
Name: Job Title, Length: 1156, dtype: object

In [17]:
df_3_abnormal['Job Title'].unique()

array(['Digital Marketing Strategist and Planner', 'Research Analyst',
       'Software Engineer-Integration', 'DevOps Engineer',
       'Real Estate Sales Agent', 'Software Quality Assurance Analyst',
       'Google Ads Specialist', 'Android Developer',
       'EY Ford Rhodes: Indirect & Direct Tax Compliance & Reporting Lead',
       'Senior Asset Integrity Engineer',
       'Senior .net core API developer', 'Senior MERN Stack Developer',
       '2D Animators', 'Mechanical/Lifting Equipment Inspector',
       'Unity  Android Mobile Game Developer', 'Junior iOS Developer',
       'Executive Chef', 'Senior Angular Developer', 'QHSE Analyst',
       'Digital Marketing Executive (Internship leading to permanent job)',
       'Tax Intern', 'Business Development training program',
       'Finance & Accounts Executive - R2R',
       'Business Development Consultant',
       'Associate, Customer Success (Sales)', 'Treasury Analyst',
       'Admissions Officer', 'Internal Auditor',
       'Ca

In [18]:
conf_titles=['Senior Asset Integrity Engineer','Business Development training program','HARO Link Builder',
             'Project Manager','CGI /Character Artist','Creative Marketing Specialist','Embedded System Engineer',
             'Enterprise Resources Planning Developer','HR Site Lead (Jamshoro)','Senior Care Quality Assurance Associate',
             'Bid Manager','Chief Technology Officer','B2B Account Executive (Sales)',
             'Technical Associate Program 2023 - Excellent Opportunity for Fresh Graduates in Pakistan']

In [19]:
# i am looking at Industry and company
df_3_abnormal[df_3_abnormal['Job Title'].isin(conf_titles)]

Job Title  \
520                     Senior Asset Integrity Engineer   
572               Business Development training program   
696                                   HARO Link Builder   
795                                     Project Manager   
798                               CGI /Character Artist   
966                       Creative Marketing Specialist   
1084                           Embedded System Engineer   
1101            Enterprise Resources Planning Developer   
1223                            HR Site Lead (Jamshoro)   
1386            Senior Care Quality Assurance Associate   
1503                                        Bid Manager   
2257                           Chief Technology Officer   
2563                      B2B Account Executive (Sales)   
2806  Technical Associate Program 2023 - Excellent O...   
5247  Technical Associate Program 2023 - Excellent O...   

                                               Job Info  \
520   Abyss Solutions Ltd · Islamabad, Islāmābād, Pa...   
572   Tech BPO (Private) Limited · Lahore, Punjab, P...   
696   HiredSupport · Lahore, Punjab, Pakistan (On-si...   
795   Raccoon Tech · Islamabad, Islāmābād, Pakistan ...   
798   Marketing Services · Islamabad, Islāmābād, Pak...   
966   Rici Melion · Lahore, Punjab, Pakistan (On-sit...   
1084  Jaffer Business Systems · Karāchi, Sindh, Paki...   
1101  Beanbend · Rawalpindi, Punjab, Pakistan (Hybri...   
1223  Haleon · Jamshoro, Sindh, Pakistan (On-site) R...   
1386  Careem · Karāchi, Sindh, Pakistan Reposted  \n...   
1503  Humantek · Karāchi, Sindh, Pakistan (On-site) ...   
2257  Giftkarte · Karāchi, Sindh, Pakistan (On-site)...   
2563  ontoedge · Rawalpindi, Punjab, Pakistan (On-si...   
2806  Siemens · Karāchi, Sindh, Pakistan (Hybrid)  \...   
5247  Siemens · Karāchi, Sindh, Pakistan Reposted  \...   

                                      Job Type  \
520               Full-time · Mid-Senior level   
572                                  Full-time   
696                                  Full-time   
795                                  Full-time   
798                                  Full-time   
966                                  Full-time   
1084                                 Full-time   
1101                                 Full-time   
1223              Full-time · Mid-Senior level   
1386              Full-time · Mid-Senior level   
1503                                 Full-time   
2257                                 Full-time   
2563                                 Full-time   
2806                   Full-time · Entry level   
5247  Hybrid\n\n \nFull-time\n\n \nEntry level   

                                             Job Link_x  \
520   https://www.linkedin.com/jobs/view/3727033391/...   
572       https://www.linkedin.com/jobs/view/3727542089   
696       https://www.linkedin.com/jobs/view/3727396682   
795       https://www.linkedin.com/jobs/view/3727544104   
798       https://www.linkedin.com/jobs/view/3727520388   
966       https://www.linkedin.com/jobs/view/3728221292   
1084      https://www.linkedin.com/jobs/view/3728221394   
1101      https://www.linkedin.com/jobs/view/3728213358   
1223      https://www.linkedin.com/jobs/view/3708206982   
1386      https://www.linkedin.com/jobs/view/3709948393   
1503      https://www.linkedin.com/jobs/view/3729936548   
2257      https://www.linkedin.com/jobs/view/3732206096   
2563      https://www.linkedin.com/jobs/view/3732810459   
2806      https://www.linkedin.com/jobs/view/3729082995   
5247      https://www.linkedin.com/jobs/view/3737472530   

                                              employees      Job ID  \
520   51-200 employees · Automation Machinery Manufa...  3727033391   
572                                    51-200 employees  3727542089   
696                                    51-200 employees  3727396682   
795                                     11-50 employees  3727544104   
798                                     11-50 employees

#### developing IT tech pattern for Filtering job titles in df_3_abnormal:

In [20]:

it_tech_pattern1 = ['Digital', 'software', 'Analyst', 'DevOps', 'Google', 'Android', '2D', 'Developer', 'Development', 'Graphic', 'Game']
it_tech_pattern2 = ['Search Engine Optimization', 'Creative', 'Video', 'Virtual', 'IT Sales', 'Application', 'Technical Content Writer']
it_tech_pattern3 = ['IT Project', 'Quality Assurance', 'Project']

it_tech_keywords = it_tech_pattern1 + it_tech_pattern2 + it_tech_pattern3
it_tech_keywords
len(it_tech_keywords)


filtered_df = df_3_abnormal[df_3_abnormal['Job Title'].apply(lambda title: any(keyword.lower() in title.lower() for keyword in it_tech_keywords))]
filtered_df
len(filtered_df)

438

#### Filtering Tech job titles in df_3_abnormal:

In [21]:
it_tech_keywords = ['Digital', 'software', 'Analyst', 'DevOps', 'Google', 'Android', '2D', 'Developer', 'Development', 'Graphic', 'Game',
                    'Search Engine Optimization', 'Creative', 'Video', 'Virtual', 'IT Sales', 'Application', 'Technical Content Writer',
                    'IT Project', 'Quality Assurance', 'Project']
it_tech_keywords
len(it_tech_keywords)

filtered_df_3_abnormal = df_3_abnormal[df_3_abnormal['Job Title'].apply(lambda title: any(keyword.lower() in title.lower() for keyword in it_tech_keywords))]
filtered_df_3_abnormal
# len(ffiltered_df_3_abnormal)

Job Title  \
23              Digital Marketing Strategist and Planner   
48                                      Research Analyst   
136                        Software Engineer-Integration   
400                                      DevOps Engineer   
435                   Software Quality Assurance Analyst   
...                                                  ...   
11680                          Quality Assurance Auditor   
11687                              Informatica Developer   
11691  Senior C++ Software Engineer, GFI Software (Re...   
11725    Software Development Support Engineer (L2 & L3)   
11732                                    Unity Developer   

                                                Job Info  \
23     Prime Human Resource Services (Pvt) Ltd · Karā...   
48     Indian School of Business · Hyderabad, Sindh, ...   
136    People (Professional Employers Pvt Ltd) · Isla...   
400    TecBrix · Lahore, Punjab, Pakistan (Remote)  \...   
435    Stewart Pakistan Private Limited · Lahore, Pun...   
...                                                  ...   
11680  US Denim Mill's · Lahore, Punjab, Pakistan  \n...   
11687  Metis International Limited · Islamabad, Islām...   
11691  Crossover  · Lahore, Punjab, Pakistan Reposted...   
11725  the ENTERTAINER · Lahore, Punjab, Pakistan  \n...   
11732  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   

                                             Job Type  \
23                                          Full-time   
48                       Full-time · Mid-Senior level   
136                                         Full-time   
400                                         Full-time   
435                                         Full-time   
...                                               ...   
11680                         On-site\n\n \nFull-time   
11687                         On-site\n\n \nFull-time   
11691   Remote\n\n \nFull-time\n\n \nMid-Senior level   
11725  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11732                         On-site\n\n \nFull-time   

                                              Job Link_x  \
23         https://www.linkedin.com/jobs/view/3727011545   
48         https://www.linkedin.com/jobs/view/3723336007   
136        https://www.linkedin.com/jobs/view/3727002795   
400    https://www.linkedin.com/jobs/view/3727012776/...   
435    https://www.linkedin.com/jobs/view/3603557857/...   
...                                                  ...   
11680      https://www.linkedin.com/jobs/view/3768907369   
11687      https://www.linkedin.com/jobs/view/3767943809   
11691      https://www.linkedin.com/jobs/view/3766356253   
11725      https://www.linkedin.com/jobs/view/3763867444   
11732      https://www.linkedin.com/jobs/view/3767951886   

                                               employees      Job ID  \
23                                     10,001+ employees  3727011545   
48                501-1,000 employees · Higher Education  3723336007   
136                               5,001-10,000 employees  3727002795   
400                                      11-50 employees  3727012776   
435                                  501-1,000 employees  3603557857   
...                                                  ...         ...   
11680                FaisaL Ahmad is hiring for this job  3768907369   
11687                                   51-200 employees  3767943809   
11691  5,001-10,000 employees · IT Services and IT Co...  3766356253   
11725           201-500 employees · Software Development  3763867444   
11732                                     1-10 employees  3767951886   

                                          Job Link_y Job Desception  \
23     https://www.linkedin.com/jobs/view/3727011545  About the job   
48     https://www.linkedin.com/jobs/view/3723336007            NaN   
136    https://www.linkedin.com/jobs/view/3727002795  About the job   
400    https://www.linkedin.com/jobs/view/37270127

### concatenating filtered_df_3_abnormal and filtered_df_5_desc:

In [22]:
filtered_df_3_Tech = pd.concat([filtered_df_5_desc,filtered_df_3_abnormal], ignore_index=True)
filtered_df_3_Tech

Job Title  \
0                               LinkedIn Data Extractor   
1                                  It Management Intern   
2                                SQL Database Developer   
3                       Business Intelligence Developer   
4                            Web Developer - Internship   
...                                                 ...   
9206                          Quality Assurance Auditor   
9207                              Informatica Developer   
9208  Senior C++ Software Engineer, GFI Software (Re...   
9209    Software Development Support Engineer (L2 & L3)   
9210                                    Unity Developer   

                                               Job Info  \
0           EcomFleet · Pakistan (Remote)  \n1 week ago   
1     Digicon Valley · Lahore, Punjab, Pakistan (On-...   
2     Murkez Technologies · Lahore, Punjab, Pakistan...   
3     DevNatives · Islamabad, Islāmābād, Pakistan (H...   
4     ItCreaters · Lahore, Punjab, Pakistan (On-site...   
...                                                 ...   
9206  US Denim Mill's · Lahore, Punjab, Pakistan  \n...   
9207  Metis International Limited · Islamabad, Islām...   
9208  Crossover  · Lahore, Punjab, Pakistan Reposted...   
9209  the ENTERTAINER · Lahore, Punjab, Pakistan  \n...   
9210  Codiea.io · Bahawalpur, Punjab, Pakistan  \n1 ...   

                                            Job Type  \
0                                          Full-time   
1                                          Full-time   
2                                          Full-time   
3                                          Full-time   
4                                          Full-time   
...                                              ...   
9206                         On-site\n\n \nFull-time   
9207                         On-site\n\n \nFull-time   
9208   Remote\n\n \nFull-time\n\n \nMid-Senior level   
9209  On-site\n\n \nFull-time\n\n \nMid-Senior level   
9210                         On-site\n\n \nFull-time   

                                         Job Link_x  \
0     https://www.linkedin.com/jobs/view/3727054368   
1     https://www.linkedin.com/jobs/view/3723395858   
2     https://www.linkedin.com/jobs/view/3727034665   
3     https://www.linkedin.com/jobs/view/3727029273   
4     https://www.linkedin.com/jobs/view/3727034696   
...                                             ...   
9206  https://www.linkedin.com/jobs/view/3768907369   
9207  https://www.linkedin.com/jobs/view/3767943809   
9208  https://www.linkedin.com/jobs/view/3766356253   
9209  https://www.linkedin.com/jobs/view/3763867444   
9210  https://www.linkedin.com/jobs/view/3767951886   

                                              employees      Job ID  \
0                                       11-50 employees  3727054368   
1                                       11-50 employees  3723395858   
2                                      51-200 employees  3727034665   
3                                       11-50 employees  3727029273   
4                   Mobeen Naqvi is hiring for this job  3727034696   
...                                                 ...         ...   
9206                FaisaL Ahmad is hiring for this job  3768907369   
9207                                   51-200 employees  3767943809   
9208  5,001-10,000 employees · IT Services and IT Co...  3766356253   
9209           201-500 employees · Software Development  3763867444   
9210                                     1-10 employees  3767951886   

                                         Job Link_y  \
0     https://www.linkedin.com/jobs/view/3727054368   
1     https://www.linkedin.com/jobs/view/3723395858   
2     https://www.linkedin.com/jobs/view/3727034665   
3     https://www.linkedin.com/jobs/view/3727029273   
4     https://www.linkedin.com/jobs/view/3727034696   
...                                             ...   
9206  https://www.linkedin.com/jobs/view/3768907369   
9

### getting non_tech_df3

In [23]:
non_tech_df3=df_3[~df_3['Job ID'].isin(filtered_df_3_Tech['Job ID'])]
non_tech_df3

Job Title  \
5      Research Analyst for Business Subjects (Office...   
6                           Performance Marketing Intern   
24                  Search Engine Optimization Executive   
26                             Machine Learning Engineer   
36                               Human Resources Trainee   
...                                                  ...   
11730                                Design Sales agents   
11731  Senior Consultant- Business Consulting- Intern...   
11734                    Information Security Specialist   
11735          FUNCTIONAL CONSULTANT ORACLE FUSION - HCM   
11736                                Community Associate   

                                                Job Info  \
5      TechoMatrix · Karāchi, Sindh, Pakistan (On-sit...   
6      8th Loop · Karāchi, Sindh, Pakistan (On-site) ...   
24     Brandix Soft  · Lahore, Punjab, Pakistan (On-s...   
26     Mexa Solutions · Lahore, Punjab, Pakistan (On-...   
36     TechSwivel (Private) Limited · Lahore, Punjab,...   
...                                                  ...   
11730  Self-employed · Pakistan  \n1 week ago\n\n · 1...   
11731  EY in Pakistan  · Karachi Division, Sindh, Pak...   
11734  DPL · Islamabad, Islāmābād, Pakistan  \n1 week...   
11735  Level 3 BOS · Karachi Division, Sindh, Pakista...   
11736  The Hive · Karachi Division, Sindh, Pakistan  ...   

                                             Job Type  \
5                                           Full-time   
6                                           Full-time   
24                                          Full-time   
26                                          Full-time   
36                                          Full-time   
...                                               ...   
11730                          Remote\n\n \nPart-time   
11731  Hybrid\n\n \nInternship\n\n \nMid-Senior level   
11734       On-site\n\n \nFull-time\n\n \nEntry level   
11735  On-site\n\n \nFull-time\n\n \nMid-Senior level   
11736                         On-site\n\n \nFull-time   

                                          Job Link_x  \
5      https://www.linkedin.com/jobs/view/3727008042   
6      https://www.linkedin.com/jobs/view/3727056183   
24     https://www.linkedin.com/jobs/view/3727025862   
26     https://www.linkedin.com/jobs/view/3727037227   
36     https://www.linkedin.com/jobs/view/3726899313   
...                                              ...   
11730  https://www.linkedin.com/jobs/view/3766852400   
11731  https://www.linkedin.com/jobs/view/3767944882   
11734  https://www.linkedin.com/jobs/view/3772530977   
11735  https://www.linkedin.com/jobs/view/3771943212   
11736  https://www.linkedin.com/jobs/view/3767944844   

                                               employees      Job ID  \
5                                        11-50 employees  3727008042   
6                                        11-50 employees  3727056183   
24                                      51-200 employees  3727025862   
26                                       11-50 employees  3727037227   
36                                       11-50 employees  3726899313   
...                                                  ...         ...   
11730              Shayan Tauqeer is hiring for this job  3766852400   
11731                 1,001-5,000 employees · Accounting  3767944882   
11734   51-200 employees · IT Services and IT Consulting  3772530977   
11735  201-500 employees · Outsourcing and Offshoring...  3771943212   
11736                                    11-50 employees  3767944844   

                                          Job Link_y  \
5      https://www.linkedin.com/jobs/view/3727008042   
6      https://www.linkedin.com/jobs/view/3727056183   
24     https://www.linkedin.com/jobs/view/3727025862   
26     https://www.linkedin.com/jobs/view/3727037227   
36     https://www.linkedin.com/jobs/view/3726899313   
...                                           

### analyzing filtered_df_3_Tech['Job Title'] and non_tech_df3['Job Title']

In [24]:
len(filtered_df_3_Tech['Job Title'].unique()),len(df_3['Job Title'].unique()),len(non_tech_df3['Job Title'].unique())

(5196, 6442, 1641)

***why there are 505 unique job titles in non_tech_df3 it should be 397 so It may be unique titles in IT and non IT jobs(mixing)***

In [25]:
5196+1641

6837

In [26]:
len(df_3['Job Title'].unique())-len(filtered_df_3_Tech['Job Title'].unique())

1246

In [27]:
6837-6442
# 2451-2343

395

***395 job titles are mixed***

### finding commen job titles

In [30]:
commen_job_titles=np.intersect1d(filtered_df_3_Tech['Job Title'].unique(),non_tech_df3['Job Title'].unique())
# type(commen_job_titles)
commen_job_titles
# len(commen_job_titles)

array(['3D Animator', '3D Graphic Designer', '3D Modeler',
       'AI Advocate II',
       'AI Architect, Trilogy (Remote) - $200,000/year USD',
       'AI Super Engineer, Skyvera (Remote) - $200,000/year USD',
       'ARCHITECTE FONCTIONNEL ASSURANCE H/F', 'ASO Expert',
       'Account Assistant', 'Account Executive', 'Account Manager',
       'Accountant', 'Accounting Associate', 'Accounts Payable Assistant',
       'Admin Officer', 'Administrative Assistant', 'Admissions Officer',
       'Android Developer', 'Angular Developer', 'Animator',
       'Appointment Setter', 'Architect', 'Architectural Drafter',
       'Area Sales Manager', 'Art Director',
       'Artificial Intelligence Engineer, Skyvera (Remote) - $200,000/year USD',
       'Artificial Intelligence Engineer, Trilogy (Remote) - $200,000/year USD',
       'Artificial Intelligence Engineer, Trilogy (Remote) - $60,000/year USD',
       'Assistant', 'Assistant Accountant', 'Assistant Engineer',
       'Assistant Manager', 'A

***There are some Tech jobs in non_tech_df and non_tech_jobs in tech_df***

In [31]:
IT_titles_in_non_tech_df3=['Android Developer','Angular Developer','Animator','Bid Manager','Closer',
                           'Computer Numerical Control Programmer','Digital Marketing Manager',
                           'Digital Marketing Specialist','Flutter Developer','Machine Learning Engineer',
                           'Senior Web Developer','Shopify Developer','Social Media Manager','Sr. Angular Developer',
                           'Technical Content Writer','Technical Project Manager','UI/UX Designer','Upwork Bidder',
                           'Virtual Customer Experience Internship','Wordpress Developer','iOS Developer']
IT_titles_in_non_tech_df3

['Android Developer',
 'Angular Developer',
 'Animator',
 'Bid Manager',
 'Closer',
 'Computer Numerical Control Programmer',
 'Digital Marketing Manager',
 'Digital Marketing Specialist',
 'Flutter Developer',
 'Machine Learning Engineer',
 'Senior Web Developer',
 'Shopify Developer',
 'Social Media Manager',
 'Sr. Angular Developer',
 'Technical Content Writer',
 'Technical Project Manager',
 'UI/UX Designer',
 'Upwork Bidder',
 'Virtual Customer Experience Internship',
 'Wordpress Developer',
 'iOS Developer']

In [32]:
filtered_df_3_Tech['Job Title'].unique()[:100]

array(['LinkedIn Data Extractor', 'It Management Intern',
       'SQL Database Developer', 'Business Intelligence Developer',
       'Web Developer - Internship', 'Sr. SDET Engineer',
       'Associate Engineer', 'Fresh Graduate Engineers',
       'Fraud Monitoring Officer', 'Business System Analyst',
       'Research Analyst - Finance', 'HR Intern',
       'Alfa Origin is looking for sales and HR Intern',
       'Sales Operations Specialist', 'ASO Internee',
       'Trainee R&D Engineer', 'Dotnet+React Developer',
       'Business Development Executive (REMOTE) up to A$2,500/month',
       'Artificial Intelligence Developer', 'Web Designer',
       'Contact Centre Rep (REMOTE) up to A$ 2,000/month',
       'Software Engineer', 'SQL Database Engineer',
       'Junior Business Developer', 'Human Resources Intern',
       'Antifraud Specialist', 'Management Trainee Officer',
       'Jr Business Developer (Upwork)',
       'Software Engineer - Product Operation',
       'International Pro

In [56]:
non_tech_df3['Job Title'].unique()[:100]

array(['Research Analyst for Business Subjects (Office Based + Night Shift)',
       'Performance Marketing Intern',
       'Search Engine Optimization Executive',
       'Machine Learning Engineer', 'Human Resources Trainee',
       'Business Development Executive',
       'Fintech B2B Marketing Executive',
       'Customer Support Representative', 'Ecommerce Manager',
       'Territory Incharge [Future roles - Multiple Locations',
       'Upwork Bidding Specialist', 'Upwork Bidder',
       'Student Leader - Chemistry', 'Customer Service Representative',
       'Business Development Specialist', 'Internal Auditor',
       'Business Development Representative-Australia region',
       'Marketing Executive', 'Business Development Representative',
       'Human Resources and Admin Executive', 'Tax Officer',
       'Sales Executive', 'Officer Sales Support - Sukkur',
       'Study Abroad Advisor', 'Customer Sales Representative',
       'Sales Officer', 'Export Operations Specialist',
   

In [33]:
filtered_df_3_Tech['Job Title'].value_counts()[:50]

Job Title
Business Development Executive                                    103
Sales Executive                                                    86
Graphic Designer                                                   58
Business Development Manager                                       57
Web Back-End Developer                                             57
Software Engineer                                                  56
Senior Software Engineer                                           43
Business Development Specialist                                    41
Full Stack Engineer                                                41
PHP Developer                                                      36
Wordpress Developer                                                36
Dotnet Developer                                                   33
Search Engine Optimization Specialist                              33
Customer Service Representative                                    32
Digital Ma

In [34]:
non_tech_df3['Job Title'].value_counts()[:50]

Job Title
Sales Executive                                                     56
Sales Specialist                                                    47
Salesperson                                                         34
Business Development Executive                                      32
Customer Sales Representative                                       23
Customer Service Representative                                     23
Business Development Specialist                                     22
Sales And Marketing Specialist                                      21
Call Center Representative                                          18
Assistant Manager Sales                                             17
Business Development Manager                                        14
Affiliate Sales (Indirect)                                          12
Assistant Manager                                                   10
Security Guard                                                     